backend/data/src/main/kotlin/io/tolgee/service/notification/EmailNotificationsService.kt

```diff
@@ -0,0 +1,117 @@
+package io.tolgee.service.notification
+
+import io.sentry.Sentry
+import io.tolgee.component.FrontendUrlProvider
+import io.tolgee.component.email.TolgeeEmailSender
+import io.tolgee.constants.NotificationType
+import io.tolgee.dtos.misc.EmailParams
+import io.tolgee.model.Notification
+import io.tolgee.model.enums.TaskType
+import io.tolgee.model.task.Task
+import io.tolgee.util.Logging
+import io.tolgee.util.logger
+import org.springframework.stereotype.Component
+
+@Component
+class EmailNotificationsService(
+  private val tolgeeEmailSender: TolgeeEmailSender,
+  private val frontendUrlProvider: FrontendUrlProvider,
+) : Logging {
+  fun sendEmailNotification(notification: Notification) {
+    val subject = composeEmailSubject(notification)
+    val text = composeEmailText(notification)
+    val params =
+      EmailParams(
+        to = notification.user.username,
+        subject = subject,
+        text =
+          """
+            |Hello!👋
+            |<br/><br/>
+            |$text
+            |<br/><br/>
+            |Regards,<br/>
+            |Tolgee
+          """.trimMargin(),
+      )
+
+    try {
+      tolgeeEmailSender.sendEmail(params)
+    } catch (e: Exception) {
+      logger.error(e.message)
+      Sentry.captureException(e)
+    }
+  }
+
+  private fun composeEmailSubject(notification: Notification) =
+    when (notification.type) {
+      NotificationType.TASK_ASSIGNED -> "New task assignment"
```

Comment:
I would move this texts to `resources/I18n_en.properties` so it's prepared for localization